unsafe_resource_interface.go

```diff
@@ -0,0 +1,18 @@
+package client
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+type unsafeResourceInterface struct {
+	client dynamic.Interface
+}
+
+func (unsaferi *unsafeResourceInterface) resourceInterface(obj *unstructured.Unstructured) (dynamic.ResourceInterface, error) {
+	plural, _ := meta.UnsafeGuessKindToResource(obj.GroupVersionKind())
```

Comment:
I'm not familiar with this. I was (perhaps naively) assuming we could use the RESTMapper for both clients.

Reply:
I found it by finding functions which returned a GroupVersionResource. I would be happy to add a RESTMapper to this, but TBH I don't know where one is in this file: https://github.com/knative/serving/blob/22f2ad98b06e86d3be708ed5dd88bfae2626ce1a/pkg/reconciler/testing/v1/factory.go#L73. I couldn't find a way to go from a Scheme to a RESTMapper. I think the reason is the RESTMapper needs the actual api endpoints and there's not a way to go from the kind to the actual resource name. My understanding is that the reason this is "Unsafe" is:

```
kind: Service
apiVersion: serving.knative.dev/v1
```

doesn't directly map to

```
services.serving.knative.dev
```

and [UnsafeGuessKindToResource](https://github.com/kubernetes/apimachinery/blob/ed135c5b96450fd24e5e981c708114fbbd950697/pkg/api/meta/restmapper.go#L123) does a "best guess that they're the same." That's my understanding. This is kinda why I think this is OK to assume for testing, but it's probably not ideal for a production usage.

Reply:
I'm just gonna trust you. :)

Reply:
![I have no idea what I'm doing](https://img.buzzfeed.com/buzzfeed-static/static/2014-10/26/6/enhanced/webdr08/longform-original-14836-1414320930-10.jpg)

Reply:
HAHAHA!

Reply:
I can make the compiler happy and validate that it works in tests. But I'm not certain this is the right way.

Reply:
I'm happy to go with this until the right way reveals itself. Will this work well enough for you that we can close https://github.com/manifestival/manifestival/pull/54?

Reply:
Yup. Closed